Checkout/Checkout/Source/Validation/Validators/CardNumberValidator.swift

```diff
@@ -89,12 +90,21 @@ final class CardNumberValidator: CardNumberValidating {
   ) -> Card.Scheme? {
     let range = NSRange(location: 0, length: cardNumber.utf16.count)
 
-    return Card.Scheme.allCases.first { scheme in
+    let matchedScheme = Card.Scheme.allCases.first { scheme in
       cardSchemeToRegex(scheme)?.firstMatch(in: cardNumber, options: [], range: range) != nil
     }
+    return addSchemePropertyIfNeeded(scheme: matchedScheme, cardNumber: cardNumber)
   }
 
   private func validateDigitsOnly(in string: String) -> Bool {
     return CharacterSet.decimalDigits.isSuperset(of: CharacterSet(charactersIn: string))
   }
+    
+  private func addSchemePropertyIfNeeded(scheme: Card.Scheme?, cardNumber: String) -> Card.Scheme? {
+    if scheme == .maestro(length: 0) {
+        return .maestro(length: cardNumber.count)
+    }
+    return scheme
```

Comment:
suggestion to link to other comment
```suggestion
    switch scheme {
    case .maestro:
        return .maestro(length: cardNumber.count)
    default:
        return scheme
    }
```